info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/view/StackedChartsView.java

```diff
@@ -383,4 +417,78 @@ protected void contributeToActionBars()
     fillLocalToolBar(bars.getToolBarManager());
   }
 
+  @SuppressWarnings("rawtypes")
+  public Object getAdapter(Class type)
+  {
+
+    if (!initEditor.get() && stackedPane.getActiveControlKey() == EDIT_VIEW)
```

Comment:
Wouldn't it be tidier to text for CommandStack first, then the initEditor, etc.

That structure would seem more consistent:
- check what type is being asked for
- try to provide a relevant object